* Ran a `centos:7` container - it was identified as `74f7f6ccdd1e`
  * Installed rsync into the container
* Inventory has a single line: `74f7f6ccdd1e`
* Playbook:

```
- hosts: all
  tasks:
    - name: Test synchronize
      synchronize: src=foo/ dest=/tmp/foo

```

* Ran:
  * `mkdir foo`
  * `touch foo/bar.txt`
  * `ansible-playbook -vvv -c docker -i inventory playbook.yml`

Output:

```
No config file found; using defaults

PLAYBOOK: playbook.yml *********************************************************
1 plays in playbook.yml

PLAY [all] *********************************************************************

TASK [Gathering Facts] *********************************************************
Using module file /private/tmp/bugfix/src/ansible/lib/ansible/modules/system/setup.py
<74f7f6ccdd1e> ESTABLISH DOCKER CONNECTION FOR USER: root
<74f7f6ccdd1e> EXEC ['/usr/local/bin/docker', 'exec', '-i', u'74f7f6ccdd1e', u'/bin/sh', '-c', u"/bin/sh -c 'echo ~ && sleep 0'"]
<74f7f6ccdd1e> EXEC ['/usr/local/bin/docker', 'exec', '-i', u'74f7f6ccdd1e', u'/bin/sh', '-c', u'/bin/sh -c \'( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1490722906.08-276721213127496 `" && echo ansible-tmp-1490722906.08-276721213127496="` echo /root/.ansible/tmp/ansible-tmp-1490722906.08-276721213127496 `" ) && sleep 0\'']
<74f7f6ccdd1e> PUT /var/folders/wg/d5z1m_tn7d31p142wgc800tw0000gp/T/tmp2SjtXE TO /root/.ansible/tmp/ansible-tmp-1490722906.08-276721213127496/setup.py
<74f7f6ccdd1e> EXEC ['/usr/local/bin/docker', 'exec', '-i', u'74f7f6ccdd1e', u'/bin/sh', '-c', u"/bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1490722906.08-276721213127496/ /root/.ansible/tmp/ansible-tmp-1490722906.08-276721213127496/setup.py && sleep 0'"]
<74f7f6ccdd1e> EXEC ['/usr/local/bin/docker', 'exec', '-i', u'74f7f6ccdd1e', u'/bin/sh', '-c', u'/bin/sh -c \'/usr/bin/python /root/.ansible/tmp/ansible-tmp-1490722906.08-276721213127496/setup.py; rm -rf "/root/.ansible/tmp/ansible-tmp-1490722906.08-276721213127496/" > /dev/null 2>&1 && sleep 0\'']
ok: [74f7f6ccdd1e]
META: ran handlers

TASK [Test synchronize] ********************************************************
task path: /private/tmp/bugfix/playbook.yml:3
The full traceback is:
Traceback (most recent call last):
  File "/private/tmp/bugfix/src/ansible/lib/ansible/executor/task_executor.py", line 125, in run
    res = self._execute()
  File "/private/tmp/bugfix/src/ansible/lib/ansible/executor/task_executor.py", line 525, in _execute
    result = self._handler.run(task_vars=variables)
  File "/private/tmp/bugfix/src/ansible/lib/ansible/plugins/action/synchronize.py", line 385, in run
    if isinstance(tmp_rsync_opts, string_types):
UnboundLocalError: local variable 'tmp_rsync_opts' referenced before assignment

fatal: [74f7f6ccdd1e]: FAILED! => {
    "failed": true, 
    "msg": "Unexpected failure during module execution.", 
    "stdout": ""
}
	to retry, use: --limit @/private/tmp/bugfix/playbook.retry

PLAY RECAP *********************************************************************
74f7f6ccdd1e               : ok=1    changed=0    unreachable=0    failed=1   

```

--- a/lib/ansible/plugins/action/synchronize.py
+++ b/lib/ansible/plugins/action/synchronize.py
@@ -382,11 +382,12 @@ class ActionModule(ActionBase):
         if self._remote_transport in [ 'docker' ]:
             # Replicate what we do in the module argumentspec handling for lists
             if not isinstance(_tmp_args.get('rsync_opts'), MutableSequence):
+                tmp_rsync_opts = _tmp_args.get('rsync_opts', '')
                 if isinstance(tmp_rsync_opts, string_types):
                     tmp_rsync_opts = tmp_rsync_opts.split(',')
                 elif isinstance(tmp_rsync_opts, (int, float)):
                     tmp_rsync_opts = [to_text(tmp_rsync_opts)]
-                _tmp_args['rsync_opts'] = tmp_rsync__opts
+                _tmp_args['rsync_opts'] = tmp_rsync_opts
 
             if '--blocking-io' not in _tmp_args['rsync_opts']:
                 _tmp_args['rsync_opts'].append('--blocking-io')

--- a/lib/ansible/plugins/action/synchronize.py
+++ b/lib/ansible/plugins/action/synchronize.py
@@ -382,7 +382,7 @@ class ActionModule(ActionBase):
         if self._remote_transport in [ 'docker' ]:
             # Replicate what we do in the module argumentspec handling for lists
             if not isinstance(_tmp_args.get('rsync_opts'), MutableSequence):
-                tmp_rsync_opts = _tmp_args.get('rsync_opts', '')
+                tmp_rsync_opts = _tmp_args.get('rsync_opts', [])
                 if isinstance(tmp_rsync_opts, string_types):
                     tmp_rsync_opts = tmp_rsync_opts.split(',')
                 elif isinstance(tmp_rsync_opts, (int, float)):

